### DISCUSSION TITLE
Strange migration behavior in drift

### DISCUSSION THREAD

**Discussion by Abbos27** [simolus3/drift#3046] [Q&A]
I wrote migration code, and everything seems to have performed well. All necessary columns and tables were added and deleted, but the schema number didn't update.

In the database file, the schema number remains at 3, but in my code, it's already 5. So, schema 3 has all the fields that were supposed to be in schema 5. Whenever the app reruns, it conflicts because the migration tries to add already existing fields.

The only workaround that came to mind was to write logic that, if `from == 3`, deletes all "migrated" code that was supposed to be added at higher schemas and then re-performs the migration.

What could cause the schema number not to update? And is there a better solution for my problem?

P.S.
I didn't perform the migration within a transaction.
The targeted device is Windows

**Comment by simolus3** [comment]
>  The targeted device is Windows

If you've tried other devices, does this only happen on Windows? I'm not sure what could cause this at the moment, did you get any exception? Drift should update the schema version right after your migration ran. If you can consistently reproduce this, does it still happen if you add `await customStatement('pragma user_version = $schemaVersion');` at the end of your migration?

**Reply by Abbos27** [reply]
Unfortunately, I can't reproduce the issue; I only have the erroneous database file. When I try to reperform the migration on this file, I get an 'SQL logic error: duplicate column name' because all the fields have already been added. Regarding the device, the app currently only works on Windows devices. Could it be a device problem that caused the issue?. I'm pretty sure the migration logic was correct, and it worked on thousands of other Windows devices, but it failed on some, adding all fields and not upgrading the schema version

**Reply by simolus3** [reply]
I wonder if the app could have crashed after applying the migration but before raising the schema version... Do you have a way to see if the devices on which this failed had other database failures from an earlier session?
